gas-tank/balance-services.md

```diff
@@ -0,0 +1,515 @@
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    owner: "string", // [!code highlight]
+    spender: "string", // [!code highlight]
+    value: "string", // [!code highlight]
+    deadline: 0, // [!code highlight]
+    v: 0, // [!code highlight]
+    r: "string", // [!code highlight]
+    s: "string" // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Balance increased successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Withdraw balance
+
+This endpoint allows users to withdraw their balance from the Gas Tank platform.
+
+To withdraw the balance, you need to sign the message with the private key of the address you want to withdraw the balance from. The `signature` field is the signature of the message signed by the address owner.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/withdraw`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json"
+    "Authorization": `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    amount: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    recipient: "string", // [!code highlight]
+    message: "string", // [!code highlight]
+    signature: "string" // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Balance withdrawn successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Internal transfer
+
+This endpoint allows users to create an internal transfer task on the Gas Tank platform.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/transfer`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    amount: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    recipient: "string", // [!code highlight]
+    message: "string", // [!code highlight]
+    signature: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Internal transfer task created successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Consume balance
+
+This endpoint allows users to consume their balance on the Gas Tank platform.
+
+To consume the balance, you need to sign the message with the private key of the address you want to consume the balance from. The `signature` field is the signature of the message signed by the address owner.
```

Comment:
Can you expand (for all the consume, deposit, withdraw) with
- What message (how to get it via which endpoint) and provide snippet
- How to sign it using ethers.js for instance (check the demo app)

Reply:
added for all @davidp94

Reply:
added for all @davidp94: https://github.com/XDeFi-tech/docs-new/pull/34/commits/b18f65e19c4eaa2a6b456b248600dd9f5a7fd901